File: automorphic.c

#include<stdio.h>
int main()
{
    int n,s=10,n1,r=0,sq=0,f=0,p=1,i;
    printf("Enter a number \n");
    scanf("%d",&n);
        sq = n * n;
        n1=n;
	while (n!=0)
	{
      r=n%10;
      f=f+1;
      n=n/10;
  }
   for(i=0;i<f;i++)
     {
      p=p*10;
     }

    if (sq%p==n1)
     printf("%d is an automorphic number",n1);
    else       
     printf("%d is not an automorphic number",n1);
     return 0;
    }
    

File: automorphic.py
n=int(input("Enter a number"))
num=n
t=10
sq=n*n
f=0
print("The square of",n,"is",sq)
while n>0 :
    r=sq%t
    if(r==num):
        f=1
        break
    n=n//10
    t=t*10
if(f==1):
    print("Automorphic number")
else:
    print("not an automorphic number")





n=int(input("Enter a number"))
num=n
sq=n*n
f=0
print("The square of",n,"is",sq)
num=str(num)
sq=str(sq)
if sq.endswith(num):
    print("Automorphic no")
else:
    print("not an automorphic no")
